### PR TITLE
server, ignore graceful shutdown timeout

### PR DESCRIPTION
**Pull request by pistej**
#239

### CODE REVIEW

**Comment by Rastusik** [comment]
@pistej https://www.conventionalcommits.org/en/v1.0.0/

**Comment by pistej** [comment]
hope commit message is ok :)

File: src/Server/HttpServer.php [hunk 1]
```diff
@@ -179,4 +183,9 @@ private function gracefulSignalShutdown(int $masterPid, float $timeoutSeconds):
             Process::kill($masterPid, $this->signalKill);
         }
     }
+
+    private function killSignalShutdown(int $masterPid): void
```
Comment (Rastusik) [inline src/Server/HttpServer.php:187]: please rename to immediateSignalShutdown

**Review by Rastusik — CHANGES_REQUESTED** [review]

File: src/Server/HttpServer.php (hunk 1 above)
Comment (pistej) [reply src/Server/HttpServer.php]: renamed

**Comment by pistej** [comment]
not sure what job `releaser-dry-run` do and if fix shoud be on my side ... ?!

**Review by Rastusik — APPROVED** [review]

**Comment by Rastusik** [comment]
@pistej no, please ignore the releaser script, it's broken and behaves non deterministically, so I was not able to fix it yet

File: tests/Feature/SwooleServerStartStopCommandTest.php [hunk 2]
```diff
@@ -61,4 +61,27 @@ public function testStartCallStopOnReactorRunningMode(): void
...
+        $serverStart->setTimeout(3);
+        $serverStart->disableOutput();
+        $serverStart->run();
+
+        $this->assertProcessSucceeded($serverStart);
+
+        $this->runAsCoroutineAndWait(function (): void {
+            $this->deferServerStop(['--no-delay']);
+
+            $client = HttpClient::fromDomain('localhost', 9999, false);
```
Comment (Rastusik) [inline tests/Feature/SwooleServerStartStopCommandTest.php:82]: I believe you have to measure how much time has passed here

**Review by Rastusik — CHANGES_REQUESTED** [review]

File: tests/Feature/SwooleServerStartStopCommandTest.php (hunk 2 above)
Comment (pistej) [reply tests/Feature/SwooleServerStartStopCommandTest.php:82]: each server shutdown takes few milisecond, not sure what to measure ... tried get somehow signal number but no success ... need help

Comment (Rastusik) [reply tests/Feature/SwooleServerStartStopCommandTest.php:82]: maybe just measure that the shutdown process took less than 1 second? or 500 ms?

**Comment by Rastusik** [comment]
@pistej can you please rebase this branch on develop? I hope I fixed the failing code coverage job

File: tests/Feature/SwooleServerStartStopCommandTest.php (hunk 2 above)
Comment (pistej) [reply tests/Feature/SwooleServerStartStopCommandTest.php]: shutdown process in booth cases takes approx 0.1 sec, .... but terminate wait until all tasks/request are done and --no-delay (kill) no, so I updated test if long request is terminated, --no-delay is called

**Comment by pistej** [comment]
rebased, seems working .... so with each push I need to wait for you to approve docker bake job and so on?? is this good solution?

**Review by Rastusik — DISMISSED** [review]

**Comment by Rastusik** [comment]
yes, sorry, it's like this. this is inherited from the original k911 repository and pipeline and it is supposed to be for security reasons (https://circleci.com/blog/protect-secrets-with-restricted-contexts/). If you would like to contribute more, I can promote you to a team member, so that way you wouldn't need to wait for the approval.

**Comment by pistej** [comment]
yea, It would be fine to allow it for me, at least for now as Im adding new features, thx. ... to see whats failed ...

**Comment by Rastusik** [comment]
@pistej done. feel free to push code directly into this repository, this way you will get rid of the need for  approval

**Review by Rastusik — APPROVED** [review]

**Comment by Rastusik** [comment]
@pistej please rebase onto current develop and I'll do the rest.

**Comment by Rastusik** [comment]
and thank you for the contribution! :)

**Comment by pistej** [comment]
done, but why still rebase ??

**Comment by Rastusik** [comment]
because this repo has linear history without merge commits and GH UI won't allow me to rebase because of conflicts. this is inherited from the original repo and since you are the first one who creates multiple requests, this has not been an issue before

**Review by Rastusik — APPROVED** [review]

**Pull request merged**